cdap-ui/app/cdap/components/DataPrep/DataPrepBrowser/DataPrepBrowserStore/ActionCreator.js

```diff
@@ -52,15 +69,59 @@ const setDatabaseAsActiveBrowser = (payload) => {
   };
   MyDataPrepApi.getConnection(params)
     .subscribe((res) => {
-      let properties = objectQuery(res, 'values', 0, 'properties');
+      let info = objectQuery(res, 'values', 0);
 
-      setDatabaseProperties({
-        properties,
-        connectionId: params.connectionId
+      MyDataPrepApi.listTables(params)
+        .subscribe((tables) => {
+          setDatabaseProperties({
+            info,
```

Comment:
Not modifying this as this will prevent repeated rendering of component subscribing to the store.